SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -311,6 +311,22 @@ def evaluate_integral(self, receiver_location, components, tolerance=1e-4):
 
         return np.vstack([rows[component] for component in components])
 
+class SimulationEquivalentSourceLayer(Simulation3DIntegral):
+    """
+    Equivalent source layer simulations
+    """
+
+    def __init__(self, mesh2D, cell_z_top, cell_z_bottom, **kwargs):
+
```

Comment:
Should probably add some docstrings on those parameters `cell_z_top` and `cell_z_bottom`.

Can likely accept `numpy.ndarray` and `float` for both?

Reply:
Can we actually make those property of the class, and do the typing and asserts in the setter?

Reply:
The properties class doesn't play nice when I tried to override with standard @property functionality. For now, I like them being required arguments.
I did make it so that you could enter a numpy array or float.

Reply:
Docstrings were also added